src/mail.c

```diff
@@ -0,0 +1,577 @@
+
+// Includes
+#include "global.h"
+#include "main.h"
+#include "overworld.h"
+#include "task.h"
+#include "unknown_task.h"
+#include "palette.h"
+#include "menu.h"
+#include "menu_helpers.h"
+#include "text.h"
+#include "text_window.h"
+#include "string_util.h"
+#include "international_string_util.h"
+#include "strings.h"
+#include "gpu_regs.h"
+#include "bg.h"
+#include "pokemon_icon.h"
+#include "species.h"
+#include "malloc.h"
+#include "easy_chat.h"
+#include "mail_data.h"
+#include "mail.h"
+
+// Static type declarations
+
+struct UnkMailStruct
+{
+    u32 numEasyChatWords:2;
+    u32 xOffset:6;
+    u32 lineHeight:8;
+};
+
+struct MailLayout
+{
+    u8 numSubStructs;
+    u8 signatureYPos;
+    u8 signatureWidth;
+    u8 wordsXPos;
+    u8 wordsYPos;
+    const struct UnkMailStruct *var8;
+};
+
+struct MailGraphics
+{
+    const u16 *palette;
+    const u8 *tiles;
+    const u8 *tileMap;
+    u16 var0C;
+    u16 var0E;
+    u16 color10;
+    u16 color12;
+};
+
+// Static RAM declarations
+
+static EWRAM_DATA struct
```

Comment:
This struct should be declared in a header elsewhere.

Reply:
I disagree.  There is exactly one object of this type anywhere in the code, and it's here.